src/meta/meta_ckpt.c

```diff
@@ -737,11 +737,42 @@ __wt_metadata_update_base_write_gen(WT_SESSION_IMPL *session, const char *config
 }
 
 /*
- * __wt_metadata_init_base_write_gen --
+ * __wt_conn_base_write_gen_correction --
+ *     Update the connection's base write generation from all files in metadata. This function must
+ *     not be called other than at the end of recovery after rollback to stable is finished.
```

Comment:
This comment helps solidify my understanding and I have a different suggestion that feels cleaner to me than the 3 functions. If this function is not called other than the end of recovery then it should be in `txn_recover.c` and named something like `__recovery_correct_write_gen`. Then it is an internal function which will call the other external functions. There is no need for this to be in the metadata code if it is specific to recovery and RTS.

Reply:
Thanks. Fixed as you suggested.